### PR TITLE
fix orchestrator nil pointer dereference error

### PR DESCRIPTION
**Pull request by QvodSoldier**
#### What type of this PR

Add one of the following kinds:
/kind bug

#### What this PR does / why we need it:
fix deploy panic when addon config is empty

#### Which issue(s) this PR fixes:
- [nil point interface](https://erda.cloud/erda/workBench/projects/387/issues/all?id=74081&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMDQzNiJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=-1&type=BUG)
- [personal mssage](https://erda.cloud/erda/workBench/projects/387/issues/all?id=73840&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMDQzNiJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=429&type=BUG)
- [job detail](https://erda.cloud/erda/workBench/projects/387/issues/all?id=64367&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMDQzNiJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=429&type=BUG)

#### Specified Reviewers:

/assign @Effet

### CODE REVIEW

**Comment by Effet** [comment]
it's bad to fix multiple bugs in one pr

File: modules/orchestrator/services/runtime/runtime.go [hunk 1]
```diff
@@ -439,7 +439,7 @@ func (r *Runtime) RedeployPipeline(operator user.ID, orgID uint64, runtimeID uin
 		logrus.Errorf(errstr)
 		return nil, err
 	}
-	dto, err := r.bdl.CreatePipeline(&apistructs.PipelineCreateRequestV2{
+	createPipelineReq := &apistructs.PipelineCreateRequestV2{
```
Comment (sfwn) [inline modules/orchestrator/services/runtime/runtime.go:442]: @chengjoey will make a new pr for this at send side.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by QvodSoldier** [comment]
> it's bad to fix multiple bugs in one pr

If split into 3 pr, I am worried that the merger cannot be completed today

**Comment by QvodSoldier** [comment]
ok, i will close this pr, Re-split into two pr

**Comment by sfwn** [comment]
> > it's bad to fix multiple bugs in one pr
> 
> If split into 3 pr, I am worried that the merger cannot be completed today

Feel free to open as many pr as possible, because we closed `update branch` a few days ago.

**Comment by Effet** [comment]
> because we closed `update branch` a few days ago.

it's also not good.

**Pull request closed without merging**